components/camel-influxdb/src/main/java/org/apache/camel/component/influxdb/InfluxDbProducer.java

```diff
@@ -138,4 +137,22 @@ private String calculateQuery(Exchange exchange) {
         return query;
     }
 
+    private void ensureDatabaseExists(String dataBaseName) {
+        QueryResult result = connection.query(new Query(SHOW_DATABASES));
+
+        //databases are in the first Serie called 'databases' and values are te fist value called 'name',
+        //if any object on the 'path' is null, database does not exist
+        boolean exists;
+        try {
+            exists = result.getResults().get(0).getSeries().get(0).getValues().get(0).contains(dataBaseName);
```

Comment:
Isn't there a nicer way than to catch a NPE

Reply:
@davsclaus  I've tried different approaches. Problem is that it is object with object with list of object with list of object and NPE could happen almost everywhere.

I'd prefer solution with streams. But it doesn't look much 'selfexplanatory'. What do you think?
>  
        dataBaseName == null || result == null ? false : result.getResults().stream()
                    .flatMap(r -> r.getSeries().stream())
                    .flatMap(s -> s.getValues() == null ? null : s.getValues().get(0).stream())
                    .anyMatch(v -> dataBaseName.equals(v));

Reply:
this could be better:

>         boolean exists = result == null ? false :
                            result.getResults() == null ? false :
                                result.getResults().get(0).getSeries() == null ? false :
                                    result.getResults().get(0).getSeries().get(0).getValues() == null ? false :
                                        result.getResults().get(0).getSeries().get(0).getValues().get(0).contains(dataBaseName);

Reply:
I've used `try { } catch` approach because it shows on the first glance what is happening. Both suggested approaches are better from my POV. But also it will take more time to understand it and it could contain some issues (like missing NPE). 
But I don't have a problem using them at all. (or different one)

Reply:
@JiriOndrusek I'm not sure, but can we use this approach https://github.com/influxdata/influxdb-java/issues/173

Reply:
@dmvolod If you are mentioning approach `influxDB.describeDatabases().contains(dbName)`, than it is unfortunatelly also depricated: https://github.com/influxdata/influxdb-java/blob/master/src/main/java/org/influxdb/InfluxDB.java#L577

Reply:
Oh, @JiriOndrusek , sure. In this case we can implement logic like origin. Thanks.

Reply:
yep lets do this